libs/ngx-sherfire/src/lib/ngx-sherfire.module.test.ts

```diff
@@ -18,183 +23,64 @@ describe(NgxSherfireModule, () => {
         storageBucket: 'storageBucket',
         firestoreEmulator: { host: 'host', port: 1234 },
         firestoreSettings: { ignoreUndefinedProperties: true },
-    };
+        appSettings: { automaticDataCollectionEnabled: false },
+    } as const satisfies FirebaseConfig;
 
-    const createAuth = jest.fn<Partial<firebase.auth.Auth>, []>();
-    const createFirestore = jest.fn<Partial<firebase.firestore.Firestore>, []>();
-    const fakeApp = {
-        auth: createAuth,
-        firestore: createFirestore,
-    } as Partial<firebase.app.App> as firebase.app.App;
+    const fakeApp: appModule.FirebaseApp = {
+        name: '[FAKE]',
+        options: {},
+        automaticDataCollectionEnabled: false,
+    };
+    const fakeAuth = {
+        app: fakeApp,
+    } as Partial<authModule.Auth> as authModule.Auth;
+    const fakeFirestore: firestoreModule.Firestore = {
+        app: fakeApp,
+        type: 'firestore',
+        toJSON() {
+            return this;
+        },
+    };
 
     beforeEach(() => {
         TestBed.configureTestingModule({ imports: [NgxSherfireModule.forRoot(config)] });
-        jest.spyOn(firebase, 'initializeApp').mockReturnValue(fakeApp);
+        jest.spyOn(appModule, 'initializeApp').mockReturnValue(fakeApp);
+        jest.spyOn(authModule, 'getAuth').mockReturnValue(fakeAuth);
+        jest.spyOn(firestoreModule, 'initializeFirestore').mockReturnValue(fakeFirestore);
     });
 
-    describe(FirebaseApp, () => {
+    describe(SherfireApp, () => {
         test('instantiation', () => {
-            expect(firebase.initializeApp).not.toBeCalled();
-            expect(TestBed.inject(FirebaseApp)).toBe(fakeApp);
-            expect(firebase.initializeApp).toBeCalledTimes(1);
-            expect(firebase.initializeApp).toBeCalledWith(config);
+            expect(appModule.initializeApp).not.toBeCalled();
+            expect(TestBed.inject(SherfireApp)).toBe(fakeApp);
+            expect(appModule.initializeApp).toBeCalledTimes(1);
+            expect(appModule.initializeApp).toBeCalledWith(config, config.appSettings);
         });
     });
 
-    describe(FirebaseAuth, () => {
+    describe(SherfireAuth, () => {
         test('instantiation', () => {
-            const theAuth = {};
-            createAuth.mockReturnValue(theAuth);
-            expect(TestBed.inject(FirebaseAuth)).toBe(theAuth);
-        });
-
-        describe('#user$', () => {
```

Comment:
The testing of user$ has all been moved to `firebase-user$.service.test.ts`, so this module test only contains tests to instantiate the different services